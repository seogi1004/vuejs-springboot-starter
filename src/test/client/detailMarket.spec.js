import { mount, shallowMount } from '@vue/test-utils'
import DetailMarketComp from '@/detailMarket'
import { samples } from './data.js'

describe('detailMarket.vue', () => {
    it('snapshot', () => {
        const cmp = mount(DetailMarketComp, {
            propsData: {
                title: 'Max Market Cap',
                data: samples[1]
            }
        });

        cmp.vm.$nextTick(() => {
            expect(cmp.vm.$el).toMatchSnapshot();
        });
    });

    it('computed', () => {
        const cmp = shallowMount(DetailMarketComp, {
            propsData: {
                title: 'Min Market Cap',
                data: samples[2]
            }
        });

        expect(cmp.vm.maxYear).toEqual('2017');
        expect(cmp.vm.minYear).toEqual('2015');
    });
});