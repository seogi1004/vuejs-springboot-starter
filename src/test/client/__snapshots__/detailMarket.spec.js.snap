// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailMarket.vue snapshot 1`] = `
<div>
  <h5>
    Max Market Cap
  </h5>
   
  <table>
    <thead>
      <tr>
        <th>
          Company
        </th>
         
        <th>
          Max Year
        </th>
         
        <th>
          Min Year
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td>
          Microsoft
        </td>
         
        <td>
          2018
        </td>
         
        <td>
          2014
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
